### ISSUE TITLE
Sourcemod Crashes upon being loaded

### ISSUE THREAD

**Issue by BreadDoesStuff** [alliedmodders/sourcemod#2065]
# Help us help you
  - [x] I have checked that my issue [doesn't exist yet](https://github.com/alliedmodders/sourcemod/issues).
  - [x] I have tried my absolute best to reduce the problem-space and have provided the absolute smallest test-case possible.
  - [x] I can always reproduce the issue with the provided description below.

# Environment
  * Operating System version: **Ubuntu Server 22.04.3 LTS**
  * Game/AppID (with version if applicable): **440**
  * Current SourceMod version: **1.11 - 6940**
  * Current SourceMod snapshot: **1.12 - 7060**
  * Current Metamod: Source snapshot: **2.0 - 1237**
  - [x] I have updated SourceMod to the [latest version](https://www.sourcemod.net/downloads.php) and it still happens.
  - [x] I have updated SourceMod to the [latest snapshot](https://www.sourcemod.net/downloads.php?branch=dev) and it still happens.
  - [x] I have updated SourceMM to the [latest snapshot](https://sourcemm.net/downloads.php?branch=dev) and it still happens.

# Description

- Anytime I go to launch my dedicated tf2 server with sourcemod & metamod installed it immediately crashes. Loading the server with only metamod installed does not present any issues. Running the server without any plugins installed also runs fine.
- I tried everything I could think of. Followed instructions as closely as possible.

# Steps to Reproduce

-  Install Metamod
- Install Sourcemod
- Start Server

# Logs
[crash_20231014203454_2.dmp](https://github.com/alliedmodders/sourcemod/files/12908029/crash_20231014203454_2.dmp)
[debug.log](https://github.com/alliedmodders/sourcemod/files/12908030/debug.log)

**Comment by cbpudding** [comment]
I second this. MetaMod seems to run fine by itself but the server crashes when SourceMod is installed. The latest snapshot for MetaMod/SourceMod does not fix the issue for the TF2 dedicated server.

| MetaMod | SourceMod | Result |
| ------- | --------- | ------ |
| 1.11-1153 | N/A | Loads |
| 1.11-1153 | 1.11-6940 | Crashes |
| 1.11-1153 | 1.12-7060 | Crashes |
| 1.12-1185 | N/A | Loads |
| 1.12-1185 | 1.12-7060 | Crashes |

**Comment by Gnomesenpai** [comment]
Also experiencing this. Using SM 1.11.0.6923 and MM 1.12.0-dev+1150 are fine though...

**Comment by BreadDoesStuff** [comment]
> Also experiencing this. Using SM 1.11.0.6923 and MM 1.12.0-dev+1150 are fine though...
> 
> Bread are you running within a Docker container? If so who made it?

Not running within any docker container. Just using a standard launch script for srcds.

**Comment by Gnomesenpai** [comment]
> > Also experiencing this. Using SM 1.11.0.6923 and MM 1.12.0-dev+1150 are fine though...
> > Bread are you running within a Docker container? If so who made it?
> 
> Not running within any docker container. Just using a standard launch script for srcds.

Been speaking to someone in discord as i've been having the same issue within a docker container, but looks like its a SoureMod issue *phew*

**Comment by BreadDoesStuff** [comment]
Just tried the versions that you said were working fine. No crashes at all! Is there a way to prevent sourcemod from automatically updating?

**Comment by Gnomesenpai** [comment]
> Just tried the versions that you said were working fine. No crashes at all! Is there a way to prevent sourcemod from automatically updating?

You can turn it off in /tf/cfg/sourcemod/sourcemod.cfg (if i remember correctly)

Looks like it may be related to this: https://github.com/alliedmodders/sourcemod/issues/2064

**Comment by sapphonie** [comment]
> > Just tried the versions that you said were working fine. No crashes at all! Is there a way to prevent sourcemod from automatically updating?
> 
> You can turn it off in /tf/cfg/sourcemod/sourcemod.cfg (if i remember correctly)
> 
> Looks like it may be related to this: https://github.com/alliedmodders/sourcemod/issues/2064

SourceMod does not auto update its own binaries. It auto updates gamedata, which you almost always want.

I'm not sure that issue is related - MM/SM should just fail to load if you're on an older operating system without a modern glibc.

In the meanwhile, could you install [Accelerator](https://forums.alliedmods.net/showthread.php?t=277703), configure it properly with your SteamID, and then post a crash link here?

**Comment by lxndr** [comment]
Same issue here. It all works fine with Debian 10. On Debian 11/12 and Ubuntu 20/22, sourcemod 1.11 crashes. I usually use docker, but I tried VM as well. Here's what I see under gdb.

![Screenshot from 2023-10-15 10-35-58](https://github.com/alliedmodders/sourcemod/assets/579459/d9d13d29-0511-42f7-a609-a735c020f082)

**Comment by Gnomesenpai** [comment]
My docker containers are based on Debian 11, this should have it in?  If not what package is required as this is a breaking change. Accelerator does not load because the server crashes before any plugins load.

**Comment by lxndr** [comment]
I believe it's a recent regression introduced in build 1.11-git6940. Because for me, the build 1.11-git6939 works fine.

**Comment by dvander** [comment]
This happened after we upgraded from clang-11 to clang-13. Psychonic managed to figure out that clang-14 fixes it, so we've landed those changes everywhere now. I did some digging tonight into what happened.

This change in clang-12 introduced a regression of some sort: https://github.com/llvm/llvm-project/commit/a084c0388e2a59b9556f2de0083333232da3f1d6

This caused SourceMod/Metamod:Source to start getting bogus PLT/GOT codegen. Effectively the PIC register was garbage. Why we were even generating GOT/PIC code is strange because we do not use -fPIC on x86.

I'll try bisecting to what fixed this for my own curiosity.
